src/Illuminate/Contracts/Container/Container.php

```diff
@@ -72,6 +72,17 @@ public function bindMethod($method, $callback);
      */
     public function bindIf($abstract, $concrete = null, $shared = false);
 
+    /**
+     * Register a binding when condition is true.
+     *
+     * @param  bool  $condition
+     * @param  string  $abstract
+     * @param  \Closure|string|null  $concrete
+     * @param  bool  $shared
+     * @return void
+     */
+    public function bindWhen($condition, $abstract, $concrete = null, $shared = false);
```

Comment:
We can't add methods to contracts on patch releases. Please remove this or send this in for master (11.x). Thanks